screen/OrderData.xml.ftl

```diff
@@ -0,0 +1,83 @@
+    <OrderTotal>${orderHeaderAndPart.partTotal}</OrderTotal>
+    <TaxAmount>XX</TaxAmount>
+    <ShippingAmount>${part.shippingCost}</ShippingAmount>
+    <CustomerNotes><![CDATA[${part.shippingInstructions}]]></CustomerNotes>
+    <InternalNotes><![CDATA[${part.shippingInstructions}]]></InternalNotes>
+    <Gift>${orderHeaderAndPart.isGift!}</Gift>
+    <GiftMessage>${orderHeaderAndPart.giftMessage!}</GiftMessage>
+    <Customer>
+      <CustomerCode><![CDATA[${orders.customer_details.email}]]></CustomerCode>
+      <BillTo>
+        <Name><![CDATA[${orders.billing_details.address.toName}]]></Name>
+        <Company><![CDATA[XX]]></Company>
+        <Phone><![CDATA[${orders.billing_details.phone.contactNumber!}]]></Phone>
+        <Email><![CDATA[${orders.customer_details.email}]]></Email>
+      </BillTo>
+      <ShipTo>
+        <#assign geo = ec.entity.find("moqui.basic.Geo").condition("geoId",shipping.address.stateProvinceGeoId!).useCache(true).one()>
+        <Name><![CDATA[${shipping.address.toName!}]]></Name>
+        <Company><![CDATA[XX]]></Company>
+        <Address1><![CDATA[${shipping.address.address1!}]]></Address1>
+        <Address2>${shipping.address.address2!}</Address2>
+        <City><![CDATA[${shipping.address.city!}]]></City>
+        <State><![CDATA[${geo.geoCodeAlpha2!}]]></State>
+        <PostalCode><![CDATA[${shipping.address.postalCode!}]]></PostalCode>
+        <Country><![CDATA[${shipping.address.countryGeoId!}]]></Country>
+        <Phone><![CDATA[${shipping.phone.contactNumber!}]]></Phone>
+      </ShipTo>
+    </Customer>
+    <Items>
+    <#list items as item>
+      <Item>
+        <#assign dimension = ec.entity.find("mantle.product.ProductUomDimension").condition("productId",item.productId).condition("uomDimensionTypeId",'Weight').one()>
+        <#assign units = ec.entity.find("moqui.basic.Uom").condition("uomId",dimension.uomId).useCache(true).one()>
+        <SKU><![CDATA[${item.sku}]]></SKU>
+        <Name><![CDATA[${item.product_name}]]></Name>
+        <#assign productContent = ec.entity.find("mantle.product.ProductContent").condition("productId",item.productId).condition("productContentTypeEnumId",'PcntImageLarge').one()>
+        <#assign url = ec.resource.getLocationReference(productContent.contentLocation)>
+        <ImageUrl><![CDATA[${url}]]></ImageUrl>
+        <Weight>${dimension.value!}</Weight>
+        <WeightUnits>${units.abbreviation!}</WeightUnits>
+        <Quantity>${item.quantity}</Quantity>
+        <UnitPrice>${item.unitAmount}</UnitPrice>
+        <Location><![CDATA[XX]]></Location>
+        <#assign featuresList = ec.service.sync().name("co.hotwax.oms.ProductServices.find#Products").parameter("productId", item.productId).call()>
+          <Options>
+          <#list featuresList.products[0].features as feature>
+            <#assign featureType = ec.entity.find("moqui.basic.Enumeration").condition("enumId",feature.type!).useCache(true).one()>
```

Comment:
Add check on enumTypeId as well. If enumId is same for different types, we may not get expected results.